lib/fastlane/plugin/notarize/actions/notarize_action.rb

```diff
@@ -97,6 +104,16 @@ def self.run(params)
         ENV.delete('FL_NOTARIZE_PASSWORD')
       end
 
+      def self.attempt_staple(package_path)
+        Actions.sh(
+          "xcrun stapler staple \"#{package_path}\"",
```

Comment:
Having two instances of this `xcrun stapler staple` sh command could be difficult to maintain down the line. Should we add a parameter to this function to indicate whether to raise or not? This way we can use the same function after a successful notarize operation as well. In this case the function would simply be called `staple`.

(Based on the parameter, I believe it's possible to re-raise the error—[here's an example](https://stackoverflow.com/a/23771227/1002763)).

Reply:
Good feedback.  As I was working on this I realized that I like removing the boolean return  value and just allow the raised error to be dealt with in the early staple situation.  Let me know what you think.